src/listeners/ticket/create-modal-submit-validator.ts

```diff
@@ -0,0 +1,183 @@
+        if (!ticketChannelId) {
+            return;
+        }
+        const ticketChannel = await guild.channels.fetch(
+            ticketChannelId
+        );
+        if (ticketChannel?.type !== ChannelType.GuildText) {
+            return;
+        }
+
+        const latestTicket = await ticketLogs.findOne(
+            { '_id.guild': interaction.guildId! },
+            { sort: { 'number': -1 } }
+        );
+
+        
+        const latestTicketNumber = latestTicket ? latestTicket.number : 0;
+        const ticketID = String(latestTicketNumber + 1).padStart(6, '0');
+
+        const threadName = `${title} - #${ticketID}`;
+
+        let thread = await ticketChannel.threads.create({
+            name: threadName,
+            reason: `Ticket for ${interaction.user.id}`,
+            type: ChannelType.PrivateThread,
+            invitable: false,
+        });
+
+        const roles = await guild.roles.fetch();
+
+        await thread.send(
+            [
+                interaction.user,
+                ...roles
+                    .filter(role =>
+                        role
+                            .permissionsIn(ticketChannel)
+                            .has(PermissionFlagsBits.ManageThreads)
+                    )
+                    .values(),
+            ].join(' ')
+        );
+
+        const ticketCreate = await thread.send({
+            embeds: [
+                new EmbedBuilder()
+                    .setColor(Color.Yellow)
+                    .setAuthor({
+                        name: interaction.user.tag,
+                        url: userUrl(interaction.user.id),
+                        iconURL: (interaction.member instanceof GuildMember
+                            ? interaction.member
+                            : interaction.user
+                        ).displayAvatarURL(),
+                    })
+                    .setTitle(`${title}`)
+                    .setDescription(exp)
+                    .setTimestamp(interaction.createdTimestamp),
+            ],
+            components: [
+                new ActionRowBuilder<ButtonBuilder>().setComponents(
+                    new ButtonBuilder()
+                        .setCustomId(TButtonId.Close)
+                        .setStyle(ButtonStyle.Danger)
+                        .setLabel('Close Ticket'),
+                ),
+            ],
+        });
+
+        messageLogger.logTicketCreation(
+            member,
+            thread,
+            title,
+            ticketID,
+            exp,
+            interaction.createdTimestamp
+        );
+
+        const ticketLogEntry: TicketLog = {
+            _id: {
+                guild: interaction.guildId!,
+                channel: thread.id,
+                user: interaction.user.id,
+            },
+            title: title,
+            number: parseInt(ticketID),
+            open: true,
+        };
+        
+        await ticketLogs.insertOne(ticketLogEntry);
```

Comment:
Can we do away with the `ticketLogs` database collection? If we keep track of the last ticket number in the `guildSettings` collection as I suggested above, it seems like the utility of this collection goes away.